src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.ons.ssdc.jobprocessor.schedule;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ssdc.common.model.entity.Job;
+import uk.gov.ons.ssdc.common.model.entity.JobRow;
+import uk.gov.ons.ssdc.common.model.entity.JobRowStatus;
+import uk.gov.ons.ssdc.common.model.entity.JobType;
+import uk.gov.ons.ssdc.common.validation.ColumnValidator;
+import uk.gov.ons.ssdc.jobprocessor.repository.JobRepository;
+import uk.gov.ons.ssdc.jobprocessor.repository.JobRowRepository;
+import uk.gov.ons.ssdc.jobprocessor.utility.JobTypeHelper;
+import uk.gov.ons.ssdc.jobprocessor.utility.JobTypeSettings;
+
+@Component
+public class RowChunkValidator {
+  private final JobRowRepository jobRowRepository;
+  private final JobRepository jobRepository;
+  private final JobTypeHelper jobTypeHelper;
+
+  public RowChunkValidator(
+      JobRowRepository jobRowRepository, JobRepository jobRepository, JobTypeHelper jobTypeHelper) {
+    this.jobRowRepository = jobRowRepository;
+    this.jobRepository = jobRepository;
+    this.jobTypeHelper = jobTypeHelper;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
+  public boolean processChunk(Job job) {
+    boolean hadErrors = false;
+
+    JobTypeSettings jobTypeSettings =
+        jobTypeHelper.getJobTypeSettings(job.getJobType(), job.getCollectionExercise());
+
+    List<JobRow> jobRows =
+        jobRowRepository.findTop500ByJobAndJobRowStatus(job, JobRowStatus.STAGED);
+
+    boolean getValidationRulesPerRow = false;
```

Comment:
Just an idea to tidy up this complex area - maybe not for now as this works, but if we stuck a boolean getValidationRulesPerRow in JobTypeHelper for each Type.  

Then here we could just have

boolean getValidationRulesPerRow = jobTypeHelper.isGetValidationPerRow();
ColumnValidator[] columnValidators =  jobTypeSettings.getColumnValidators();  //If this is null, nothing lost

Reply:
I had some thoughts about more major refactoring... but decided that it'd be better on another card